presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeProbabilityFunctionQueries.java

```diff
@@ -23,10 +23,23 @@
 public abstract class AbstractTestNativeProbabilityFunctionQueries
         extends AbstractTestQueryFramework
 {
+    // The superclass method returns an ExpectedQueryRunner,
+    // but this class expects a more specific subclass - QueryRunner.
+    // Make this requirement explicit
+    @Override
```

Comment:
Why have you added this ?

Other classes like AbstractTestNativeAggregations.java don't have these functions and simply state 
QueryRunner queryRunner = (QueryRunner) getExpectedQueryRunner(); in the createTables  function.

Reply:
This abstract class has some expectations of its subclasses:
- `createExpectedQueryRunner` must be overridden
- `createExpectedQueryRunner` must return an object of type `QueryRunner`.

Unless these requirements are satisfied, the following line will throw a Runtime Exception:
`QueryRunner queryRunner = (QueryRunner) getExpectedQueryRunner()`

This abstract function formalises the above two requirements. If a subclass fails to satisfy the requirement, there will be a compile-time error.

In short, adding this takes advantage of Java's type checking and turns potential runtime errors into compile-time errors, which will be noticed and fixed earlier in the development cycle.

Reply:
Ah I see. So the AbstractTestQueryFramework was written for presto-java tests that compare PrestoJava (has QueryRunner) with H2QueryRunner (has ExpectedQueryRunner). In presto-native tests, presto_cpp (has QueryRunner) is compared with PrestoJava(has QueryRunner). As the presto-java side is used to create tables (since that support is limited in C++) so `QueryRunner queryRunner = (QueryRunner) getExpectedQueryRunner()` works implicitly. You are making this explicit. 

While this is a good idea, it might be better to do it at a more generic level in presto-native-execution than specifically for this class. I would derive AbstractNativeTestQueryFramework from AbstractTestQueryFramework and change the implementation.

But lets not complicate this here I feel. We can follow the same style as other native tests for now. If there is strong feeling to change ExpectedQueryRunner to QueryRunner for the java query runner we can do it with a generic solution across all tests.

Reply:
That's fair. I've removed the method overrides, it should now match the other test classes.